### PR TITLE
[JENKINS-34251] Some additional tests

### PR DESCRIPTION
**Pull request by andresrc**
See [JENKINS-34251](https://issues.jenkins-ci.org/browse/JENKINS-34251)

Small additions so that for both scripts and classpath entries we have:
* Configure and use with and without security.
* Full cycle of several entities checking from the model and from the UI.

Most of the changes are to factor out the common parts.

@reviewbybees

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/scriptsecurity/scripts/AbstractApprovalTest.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2014 Jesse Glick, CloudBees Inc.
```
Comment (recena) [inline src/test/java/org/jenkinsci/plugins/scriptsecurity/scripts/AbstractApprovalTest.java]: No updated

Comment (andresrc) [inline src/test/java/org/jenkinsci/plugins/scriptsecurity/scripts/AbstractApprovalTest.java]: mmmm, a mix of renamed and modified files.... will polish...

File: src/main/resources/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval/index.jelly [hunk 2]
```diff
@@ -185,15 +185,15 @@ THE SOFTWARE.
                     <j:forEach var="ps" items="${it.pendingScripts}">
                         <div id="ps-${ps.hash}" class="pending-script">
                             <p class="ps-context">
-                                <button onclick="approveScript('${ps.hash}')">Approve</button> / <button onclick="denyScript('${ps.hash}')">Deny</button> ${ps.language.displayName} script
+                                <button class="approve" onclick="approveScript('${ps.hash}')">Approve</button> / <button class="deny" onclick="denyScript('${ps.hash}')">Deny</button> ${ps.language.displayName} script
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval/index.jelly]: Could probably be useful in `acceptance-test-harness` too, right?

File: src/test/java/org/jenkinsci/plugins/scriptsecurity/scripts/AbstractApprovalTest.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    @Test public void withSecurityRejected() throws Exception {
+        configureSecurity();
+        create().assertCannotUse();
+    }
+
+    @Test public void withSecurityAccepted() throws Exception {
+        configureSecurity();
+        create().assertCannotUse().approve().use();
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/scriptsecurity/scripts/AbstractApprovalTest.java]: :ant: Does this not subsume `withSecurityRejected`, increasing test time to no purpose?

**Comment by jglick** [comment]
:bee:

File: src/main/resources/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval/index.jelly [hunk 4]
```diff
@@ -185,15 +185,15 @@ THE SOFTWARE.
...
                                 <st:include it="${ps.context}" page="index.jelly"/>:
                             </p>
                             <f:textarea readonly="readonly" codemirror-mode="${ps.language.codeMirrorMode}" codemirror-config='"readOnly": true' rows="10" cols="80" value="${ps.script}"/>
                         </div>
                     </j:forEach>
                 </j:otherwise>
             </j:choose>
-            <p>
+            <p id="approvedScripts-clear">
```
Comment (recena) [inline src/main/resources/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval/index.jelly]: It is not a good practice to use CamelCase, better `approvedscripts-clear`.

**Pull request merged**